File: ios/FieldViewManager.m
#import <Foundation/Foundation.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(FieldViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(labelText, NSString)
RCT_EXPORT_VIEW_PROPERTY(placeholderText, NSString)
RCT_EXPORT_VIEW_PROPERTY(formLabel, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(formControl, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(errorMessage, NSDictionary)

@end

File: ios/FieldViewManager.swift
import Foundation

@objc(FieldViewManager)
class FieldViewManager: RCTViewManager {
    override func view() -> UIView! {
        return FieldView()
    }
  
    override class func requiresMainQueueSetup() -> Bool {
    return false
    }

    @objc func setFormLabel(_ node: NSNumber, formLabel: NSDictionary) {
        DispatchQueue.main.async { [weak self] in
            if let field = self?.bridge.uiManager.view(forReactTag: node) as? FieldView {
                field.formLabel = formLabel
            }
        }
    }

    @objc func setFormControl(_ node: NSNumber, formControl: NSDictionary) {
        DispatchQueue.main.async { [weak self] in
            if let field = self?.bridge.uiManager.view(forReactTag: node) as? FieldView {
                field.formControl = formControl
            }
        }
    }

    @objc func setErrorMessage(_ node: NSNumber, errorMessage: NSDictionary) {
        DispatchQueue.main.async { [weak self] in
            if let field = self?.bridge.uiManager.view(forReactTag: node) as? FieldView {
                field.errorMessage = errorMessage
            }
        }
    }
}
